lua/class_definitions.lua

```diff
@@ -75,6 +75,24 @@ classes = {
         functions = {
         }
     },
+    calendar = {
+        attributes = {
+        },
+        functions = {
+            { name = "get_turn", rval = "int", args = { } },
+            { name = "minutes_past_midnight", rval = "int", args = { } },
+            { name = "seconds_past_midnight", rval = "int", args = { } },
+            { name = "sunrise", rval = "calendar", args = { } },
+            { name = "sunset", rval = "calendar", args = { } },
+            { name = "is_night", rval = "bool", args = { } },
+            { name = "sunlight", rval = "calendar", args = { } },
```

Comment:
Those functions here return a `calendar` object by value, but the Lua wrapper stores them all as references (se header of lua/class_definitions.lua). When the functions get called from Lua, the wrapper stores a reference to a temporary object in Lua, which is immediately destroy, leaving an invalid reference behind.


`get_calendar_turn_wrapper` is fine, it returns a reference to the global calendar object.